policybot/cmd/lifecycle.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"istio.io/bots/policybot/pkg/gh"
+
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/grpclog"
+
+	"istio.io/bots/policybot/pkg/config"
+	"istio.io/bots/policybot/pkg/lifecycler"
+	"istio.io/bots/policybot/pkg/storage/spanner"
+	"istio.io/pkg/env"
+	"istio.io/pkg/log"
+)
+
+func lifecycleCmd() *cobra.Command {
```

Comment:
This file seems 90% identical to syncer.  Is there a way to give them both an implicit parent command to avoid duplication?

Reply:
Yeah, I guess I can create some shared code that deals with all these common arguments.

Reply:
Done. PTAL.